lib/driver.js

```diff
@@ -484,6 +482,52 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Handles MJPEG server-related capabilities
+   * @returns {Promise<void>}
+   */
+  async handleMjpegOptions() {
+    await this.allocateMjpegServerPort();
+    // turn on mjpeg stream reading if requested
+    if (this.opts.mjpegScreenshotUrl) {
+      this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+      this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+      await this.mjpegStream.start();
+    }
+  }
+
+  /**
+   * Allocates and configures port forwarding for the MJPEG server
+   * @returns {Promise<void>}
+   * @throws {Error} If port forwarding fails
```

Comment:
If port forwarding fails and `mjpegServerPort` capability value is provided explcitly